code/game/objects/items/stacks/medical.dm

```diff
@@ -81,8 +76,18 @@
 			M.updatehealth()
 		else
 			user << "<span class='notice'>Medicine won't work on a robotic limb!</span>"
+			return
 	else
+		var/t_himself = "itself"
+		if(user.gender == MALE)
+			t_himself = "himself"
+		else if(user.gender == FEMALE)
+			t_himself = "herself"
+		user.visible_message("<span class='notice'>[user] starts to apply [src] on [t_himself]...</span>", "<span class='notice'>You begin applying [src] on yourself...</span>")
 		M.heal_organ_damage((src.heal_brute/2), (src.heal_burn/2))
+		if(!do_mob(user, M, self_delay))	return
+		user.visible_message("<span class='green'>[user] applies [src] on [t_himself].</span>", "<span class='green'>You apply [src] on yourself.</span>")

```

Comment:
What the hell? You're putting the "heal yourself" code in the "target is not human" block and removing it from the right place? For what purpose?